python/ray/tests/test_ray_shutdown.py

```diff
@@ -475,5 +475,32 @@ def verify():
     wait_for_condition(verify)
 
 
+@pytest.mark.skipif(platform.system() == "Windows", reason="Hang on Windows.")
+def test_kill_actor_after_restart(shutdown_only):
+    """Test that killing an actor from a previous session raises a helpful error."""
+    # Set include_dashboard=False to have faster startup.
+    ray.init(num_cpus=1, include_dashboard=False)
+
+    @ray.remote
+    class A:
+        pass
+
+    a = A.remote()
+
+    # Restart ray
+    ray.shutdown()
+    ray.init(num_cpus=1, include_dashboard=False)
+
+    # Attempting to kill an actor from the previous session should raise
+    # a helpful error message instead of crashing the interpreter.
+    with pytest.raises(
+        ValueError, match="ActorHandle objects are not valid across Ray sessions"
```

Comment:
Since now worker raised `ActorHandleNotFoundError`, should we not assert that instead of ValueError?